src/SWP/Bundle/CoreBundle/Context/ScopeContext.php

```diff
@@ -14,19 +14,66 @@
 
 namespace SWP\Bundle\CoreBundle\Context;
 
+use SWP\Bundle\CoreBundle\Theme\Model\ThemeInterface;
 use SWP\Bundle\SettingsBundle\Context\ScopeContext as BaseScopeContext;
+use SWP\Bundle\SettingsBundle\Model\SettingsOwnerInterface;
+use SWP\Component\MultiTenancy\Context\TenantContextInterface;
+use Sylius\Bundle\ThemeBundle\Context\ThemeContextInterface;
 
 class ScopeContext extends BaseScopeContext implements ScopeContextInterface
 {
-    /**
-     * {@inheritdoc}
-     */
+    const NEW_SCOPES = [
+        ScopeContextInterface::SCOPE_ORGANIZATION,
+        ScopeContextInterface::SCOPE_TENANT,
+        ScopeContextInterface::SCOPE_THEME,
+    ];
+
+    private $tenantContent;
+
+    private $themeContext;
+
+    public function __construct(TenantContextInterface $tenantContext, ThemeContextInterface $themeContext)
+    {
+        $this->tenantContent = $tenantContext;
+        $this->themeContext = $themeContext;
+    }
+
     public function getScopes(): array
     {
-        return array_merge(parent::getScopes(), [
-            ScopeContextInterface::SCOPE_ORGANIZATION,
-            ScopeContextInterface::SCOPE_TENANT,
-            ScopeContextInterface::SCOPE_THEME,
-        ]);
+        return array_merge(parent::getScopes(), self::NEW_SCOPES);
+    }
+
+    public function getScopesOwners(): array
+    {
+        foreach ($this->getScopes() as $scope) {
+            $this->lazyLoad($scope);
+        }
+
+        return parent::getScopesOwners();
+    }
+
+    public function getScopeOwner(string $scope)
+    {
+        $this->lazyLoad($scope);
+
+        return parent::getScopeOwner($scope);
+    }
+
+    private function lazyLoad(string $scope)
+    {
+        if (in_array($scope, self::NEW_SCOPES)) {
+            $tenant = $this->tenantContent->getTenant();
+            if ($tenant instanceof SettingsOwnerInterface) {
+                if (ScopeContextInterface::SCOPE_TENANT === $scope) {
```

Comment:
can this be somehow simplified? It's really hard to read that, what about using a switch statement instead?